src/Bathymetry.jl

```diff
@@ -189,26 +179,44 @@ function regrid_bathymetry(target_grid;
 
     target_z = interpolate_bathymetry_in_passes(native_z, target_grid; 
                                                 passes = interpolation_passes,
-                                                connected_regions_allowed,
+                                                major_basins,
                                                 minimum_depth)
 
+    if minimum_depth > 0
+        zi = interior(target_z, :, :, 1)
+        shallow_ocean = zi .> - minimum_depth
+        zi[shallow_ocean] .= 0
+    end
+
+    if major_basins < Inf
+        remove_minor_basins!(target_z, major_basins)
+    end
+
+    fill_halo_regions!(target_z)
+
     return target_z
 end
 
 # Here we can either use `regrid!` (three dimensional version) or `interpolate`
 function interpolate_bathymetry_in_passes(native_z, target_grid; 
                                           passes = 10,
-                                          connected_regions_allowed = 3,
+                                          major_basins = Inf,
                                           minimum_depth = 0)
     Nλt, Nφt = Nt = size(target_grid)
```

Comment:
I thought that `size` would always return a 3-tuple. How does this work?

Reply:
`Nλt, Nφt` are the first two arguments of `size`. If we want to make it more explicit, `Nλt, Nφt, _` is ok too